pkg/actions/addon/create.go

```diff
@@ -31,13 +31,16 @@ func (a *Manager) Create(ctx context.Context, addon *api.Addon, waitTimeout time
 	// First check if the addon is already present as an EKS managed addon
 	// and if so, don't re-create
 	var notFoundErr *ekstypes.ResourceNotFoundException
-	_, err := a.eksAPI.DescribeAddon(ctx, &eks.DescribeAddonInput{
+	summary, err := a.eksAPI.DescribeAddon(ctx, &eks.DescribeAddonInput{
 		AddonName:   &addon.Name,
 		ClusterName: &a.clusterConfig.Metadata.Name,
 	})
 	if err == nil {
-		logger.Info("Addon %s is already present in this cluster, as an EKS managed addon, and won't be re-created", addon.Name)
-		return nil
+		// we should try to re-create the addon if the existing one has CREATE_FAILED status
+		if summary.Addon.Status != ekstypes.AddonStatusCreateFailed {
+			logger.Info("Addon %s is already present in this cluster, as an EKS managed addon, and won't be re-created", addon.Name)
+			return nil
```

Comment:
this is a bit confusing, comment and logger seem to suggest different things. 
```
logger.Info("Addon %s is already present in this cluster, as an EKS managed addon, and won't be re-created", addon.Name)
		return nil
``` 
I think this is fine as it is.

Reply:
I have refactored this code so that it's easier to understand and the comments match the functionality
https://github.com/weaveworks/eksctl/pull/5880/commits/c22f1eeab6b92770be0b78a8555e2ee868bc87fe